bgpd/bgp_route.c

```diff
@@ -3901,10 +3901,10 @@ bgp_static_update (struct bgp *bgp, struct prefix *p,
 
 	  /* Nexthop reachability check. */
          if (bgp_flag_check (bgp, BGP_FLAG_IMPORT_CHECK) &&
-              safi == SAFI_UNICAST)
+             (safi == SAFI_UNICAST || safi == SAFI_LABELED_UNICAST))
 	    {
 	      if (bgp_find_or_add_nexthop (bgp, afi, ri, NULL, 0) &&
-                  safi == SAFI_UNICAST)
+                  (safi == SAFI_UNICAST || safi == SAFI_LABELED_UNICAST))
```

Comment:
Isn't this SAFI check here superfluous? If the if statement above returns true we already know that safi is either `SAFI_UNICAST` or `SAFI_LABELED_UNICAST`.

Also, there's an additional check for `BGP_FLAG_IMPORT_CHECK` a few lines down in the same function, when a route is not found and we need to create a new one. I believe we need to check the SAFI there too.